tuna.go

```diff
@@ -311,6 +314,33 @@ func (c *Common) CreateServerConn(force bool) error {
 
 				metadata := c.GetMetadata()
 
+				if filter != nil && !filter.Empty() {
```

Comment:
You can change the `Empty()` function so it can be called even when filter is nil:
```
function (filter *IPFIlter) Empty() bool {
  if filter == nil {
    return true
  }
  ...
}
```